src/Build.UnitTests/BackEnd/TaskHostFactory_Tests.cs

```diff
@@ -36,8 +36,15 @@ public void TaskNodesDieAfterBuild()
                 string.IsNullOrEmpty(processId).ShouldBeFalse();
                 Int32.TryParse(processId, out int pid).ShouldBeTrue();
                 Process.GetCurrentProcess().Id.ShouldNotBe<int>(pid);
-                Thread.Sleep(1000);
-                Should.Throw<ArgumentException>(() => Process.GetProcessById(pid));
+                try {
+                    Process taskHostNode = Process.GetProcessById(pid);
+                    taskHostNode.WaitForExit(2000);
+                    Should.Throw<ArgumentException>(() => Process.GetProcessById(pid));
+                }
+                catch (Exception e)
+                {
+                    e.ShouldBeOfType<ArgumentException>();
+                }
```

Comment:
Why this catch?

Reply:
We want taskHostNode to die. If it dies before this try statement, `Process.GetProcessById` will throw an ArgumentException, so this test succeeded if either the `Should.Throw` statement succeeds or the node dies fast, and `Process.GetProcessById` throws an ArgumentException.

Reply:
Comment with that reasoning, catch only the exact exception, minimize the scope of the catch, and can you assert details of it?